File: CinemaApp/src/app/app-logic/seat.ts
export class Seat{
    row:string;
    number:number;
    occupied:boolean;
    selected:boolean;

    constructor(row:string,number:number,occupied:boolean=false){
        this.row=row;
        this.number=number;
        this.occupied=occupied;
        this.selected=false;
    }

    toggleSelection():void{
        if(!this.occupied){
            this.selected=!this.selected;
        }
    }

    get id():string{
          return `${this.row}-${this.number}`;
    }
}

File: CinemaAPI/CinemaAPI/Models/Seat.cs
namespace CinemaAPI.Models
{
    public class Seat
    {
        public int Id { get; set; }
        public int Row {  get; set; }
        public int Number {  get; set; }
        public bool IsReserved {  get; set; }
        public ScreeningRoom ScreeningRoom { get; set; }

        public List<Reservation> Reservations { get; set; }

    }
}
